pkgs/applications/editors/pulsar/default.nix

```
@@ -0,0 +1,180 @@
+{ lib
+, stdenv
+, git
+, runtimeShell
+, fetchzip
+, wrapGAppsHook
+, glib
+, gtk3
+, atomEnv
+, xorg
+, libxkbcommon
+, hunspell
+, hunspellDicts
+, useHunspell ? true
+, languages ? [ "en_US" ]
+, nodePackages # We need ASAR to unpack the app.asar file and patch paths
+, python3 # Cursed patch to allow asar unpacking, as the builtin in the unpacked asar isn't there
+, makeDesktopItem
+}:
+
+let
+  owner = "pulsar-edit";
+  pname = "Pulsar";
+  version = "1.103.0";
+
+  sourcesPath = {
+    x86_64-linux = {
+      tarname = "Linux.${pname}-${version}.tar.gz";
+      hash = "sha256-O+mekV2h3mxRPYpIrpoPHQyRDuXgl+En8n8u2yBG8TQ=";
+    };
+    aarch64-linux = {
+      tarname = "ARM.Linux.${pname}-${version}-arm64.tar.gz";
+      hash = "sha256-gajIHUJrjCogcJjJVSKz85/x9inuS9rEMYlLqcolUqg=";
+    };
+  }.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+  additionalLibs = lib.makeLibraryPath [
+    xorg.libxshmfence
+    libxkbcommon
+    xorg.libxkbfile
+  ];
+  newLibpath = "${atomEnv.libPath}:${additionalLibs}";
+
+  # Hunspell for x86_64-linux
+  buildLocalePath = path: "searchPaths.push('${path}');";
+  localeDerivations = builtins.map (lang: hunspellDicts.${lang}) languages;
+  localePatchs = lib.concatMapStringsSep "" buildLocalePath localeDerivations;
+
+  # Hunspell for aarch64-linux
+  hunspellDirs = builtins.map (lang: "${hunspellDicts.${lang}}/share/hunspell") languages;
+  hunspellTargetDirs = "$out/opt/Pulsar/resources/app.asar.unpacked/node_modules/spellchecker/vendor/hunspell_dictionaries";
+  hunspellCopyCommands = lib.concatMapStringsSep "\n" (lang: "cp -r ${lang}/* ${hunspellTargetDirs};") hunspellDirs;
+in
+stdenv.mkDerivation rec {
+  inherit pname version;
+
+  src = with sourcesPath; fetchzip {
+    url = "https://github.com/${owner}/${pname}/releases/download/v${version}/${tarname}";
+    inherit hash;
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook # Fix error: GLib-GIO-ERROR **: No GSettings schemas are installed on the system
```

```suggestion
    wrapGAppsHook
```
we don't need to comment that, it is well known that apps using gnome libs require this hook